Editor/Scripts/CodeGenerators/RGActionCodeGenerator.cs

```diff
@@ -100,6 +106,103 @@ private static void SearchForBotActionMethods()
+
+                foreach (var classDeclaration in classDeclarations)
+                {
+                    string className = classDeclaration.Identifier.ValueText;
+                    List<RGStateInfo> stateList = new List<RGStateInfo>();
+
+                    var membersWithRGState = classDeclaration.Members
+                        .Where(m => m.AttributeLists.Any(a => a.Attributes.Any(attr => attr.Name.ToString() == "RGState")));
+
+                    foreach (var member in membersWithRGState)
+                    {
+                        string fieldType = member is MethodDeclarationSyntax ? "method" : "variable";
+                        string fieldName = member is MethodDeclarationSyntax
+                            ? ((MethodDeclarationSyntax) member).Identifier.ValueText
+                            : ((FieldDeclarationSyntax) member).Declaration.Variables.First().Identifier.ValueText;
+
+                        string stateName = fieldName;
+                        var attribute = member.AttributeLists.SelectMany(attrList => attrList.Attributes)
+                            .FirstOrDefault(attr => attr.Name.ToString() == "RGState");
+
+                        if (attribute != null)
+                        {
+                            var attributeArgument = attribute.ArgumentList?.Arguments.FirstOrDefault();
+                            if (attributeArgument != null &&
+                                attributeArgument.Expression is LiteralExpressionSyntax literal)
+                            {
+                                stateName = literal.Token.ValueText;
+                            }
+                        }
+
+                        string type = member is MethodDeclarationSyntax
+                            ? RemoveGlobalPrefix(semanticModel.GetTypeInfo(((MethodDeclarationSyntax) member).ReturnType)
+                                .Type.ToDisplayString(SymbolDisplayFormat.FullyQualifiedFormat))
+                            : RemoveGlobalPrefix(semanticModel.GetTypeInfo(((FieldDeclarationSyntax) member).Declaration.Type)
+                                .Type.ToDisplayString(SymbolDisplayFormat.FullyQualifiedFormat));
+
+                        stateList.Add(new RGStateInfo
+                        {
+                            FieldType = fieldType,
+                            FieldName = fieldName,
+                            StateName = stateName,
+                            Type = type
+                        });
+                    }
+
+                    if (stateList.Any())
+                    {
+                        rgStateInfoList.Add(new RGStatesInfo
+                        {
+                            Object = className,
+                            State = stateList
+                        });
+                    }
+                }
+            }
+
+            string jsonResult = JsonConvert.SerializeObject(new { RGStateInfo = rgStateInfoList }, Formatting.Indented);
```

Comment:
Seems like we are serializing just to deserialize in the generate call? Is this because we eventually plan to send it somewhere?

Reply:
I guess this is the spot where we could write it and then that is what Abby will be able to upload?

Reply:
Yeah exactly - I'm serializing to export it as a `.json` file, so they can upload it for @abeizer 's work